packages/contract/src/EphemeralNotes.sol

```diff
@@ -0,0 +1,57 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity ^0.8.12;
+
+import "openzeppelin-contracts/contracts/token/ERC20/IERC20.sol";
+import "openzeppelin-contracts/contracts/utils/cryptography/ECDSA.sol";
+
+struct Note {
+    address ephemeralOwner;
+    address from;
```

Comment:
storing `from` for convenience LGTM. it's redundant with logs, but L2 state is very cheap and the expensive part (calldata) is same either way.

Reply:
Yeah, it's also stored so we are able to support the edge case of sender reverting unclaimed notes but the sender has lost the ephemeral private key themselves (can happen if for instance user deletes app and comes back years later or something)